src/Message.cpp

```diff
@@ -616,6 +616,120 @@ bool Message::isEmpty() const
     return sd_bus_message_is_empty((sd_bus_message*)msg_) != 0;
 }
 
+pid_t Message::getCredsPid() const
+{
+    uint64_t mask = SD_BUS_CREDS_PID | SD_BUS_CREDS_AUGMENT;
+    sd_bus_creds *creds = nullptr;
+    int r = sdbus_->sd_bus_query_sender_creds((sd_bus_message*)msg_, mask, &creds);
+    SDBUS_THROW_ERROR_IF(r < 0, "Failed to get bus creds", -r);
+
+    pid_t pid = 0;
+    r = sdbus_->sd_bus_creds_get_pid(creds, &pid);
+    sd_bus_creds_unref(creds);
```

Comment:
I'd suggest to place unreffing into a `SCOPE_EXIT` macro in all these new methods, and place it right after `sd_bus_query_sender_creds()`. See a comment below explaining more on the macro.

Reply:
Thanks for the tip, fixed.